go/perf/suite/suite.go

```diff
@@ -464,20 +481,13 @@ func (suite *PerfSuite) getGitHead(dir string) string {
 	return strings.TrimSpace(stdout.String())
 }
 
-func (suite *PerfSuite) startServer() (host string, stopFn func()) {
-	// This is the temporary database for tests to use.
-	//
-	// * Why not use a local database + memory store?
-	// Firstly, because the spec would be "mem", and the spec library doesn't know how to reuse stores.
-	// Secondly, because it's an unrealistic performance measurement.
-	//
-	// * Why use a remote (HTTP) database?
-	// It's more realistic to exercise the HTTP stack, even if it's just talking over localhost.
-	//
-	// * Why provide an option for leveldb vs memory underlying store?
-	// Again, leveldb is more realistic than memory, and in common cases disk space > memory space.
-	// However, on this developer's laptop, there is actually very little disk space, and a lot of memory;
-	// plus making the test run a little bit faster locally is nice.
+// StartRemoteDatabase creates a new remote database on an arbitrary free port,
+// running on a separate goroutine. Returns the hostname that that database was
+// started on, and a callback to run to shut down the server.
+//
+// If the -perf.mem flag is specified, the remote database is hosted in memory,
+// not on disk (in a temporary leveldb directory).
+func (suite *PerfSuite) StartRemoteDatabase() (host string, stopFn func()) {
```

Comment:
It was deliberate but now that I look again it would make sense to all be in the docs, I think.